data/src/main/java/com/android254/data/dao/SessionDao.kt

```diff
@@ -23,5 +23,5 @@ import com.android254.data.db.model.Session
 interface SessionDao : BaseDao<Session> {
 
     @Query("SELECT * FROM SESSION")
-    fun fetchSessions(): List<Session>
+    suspend fun fetchSessions(): List<Session> // This should be suspended or return FLow [Note: Don't suspend Flows]
```

Comment:
Remove comment.

Reply:
@anselmoalexandre We should not suspend these queries. I think best way is to return flows instead

Reply:
@jumaallan any reason why we shouldn't suspend it?

Reply:
@anselmoalexandre - maybe this explains it better - https://stackoverflow.com/a/68750634

> Flows are different - their whole point is to provide the results at arbitrary times, and potentially without ever stopping! The data items are delivered whenever they're emitted, instead of all at once in a collection.

I am just thinking we should prefer using flows instead?

Reply:
> @anselmoalexandre - maybe this explains it better - https://stackoverflow.com/a/68750634
> 
> > Flows are different - their whole point is to provide the results at arbitrary times, and potentially without ever stopping! The data items are delivered whenever they're emitted, instead of all at once in a collection.
> 
> I am just thinking we should prefer using flows instead?

Yep, Flows are great. But you also need to question yourself: is it needed to use Flow? 🤔 Or I can just suspend this. 😄

Reply:
Not sure what everyone else thinks here (I would prefer using flows 😸 )

@michaelbukachi @wangerekaharun @misshannah @chepsi

Reply:
I  support @jumaallan idea of using a flow.To better support  this I can attach this link for reference [https://medium.com/androiddevelopers/room-flow-273acffe5b57](https://medium.com/androiddevelopers/room-flow-273acffe5b57)